### PR TITLE
chore (ai): rename roundtrips to steps

### PR DESCRIPTION
**Pull request by lgrammel**
## Summary
- changes `maxToolRoundtrips` parameter to `maxSteps` (for `generateText`, `streamText`)
- add `ai.settings.maxSteps` telemetry to `streamText` (and rename for `generateText`)
- **breaking change**: `roundtrip-finish` stream part from `streamText` is now called `step-finish`
- align Svelte example URL with other examples

## Background
The term `roundtrips` was idiosyncratic and had semantic issues (what is a roundtrip? how does the roundtrip number map to the `roundtrip` object size?). We are renaming to steps which has a clear meaning and maps correctly the the executed iterations.

## Tasks

- [x] generateText
- [x] generateText docs
- [x] streamText
- [x] streamText docs
- [x] rename stream part
- [x] useChat react
- [x] useChat svelte x2
- [x] useChat solid
- [x] useChat docs
- [x] general docs search
- [x] examples
   - [x] core
   - [x] react
   - [x] svelte
   - [x] solid
- [x] changeset

### CODE REVIEW

File: content/docs/07-reference/ai-sdk-core/01-generate-text.mdx [hunk 1]
```diff
@@ -347,11 +347,11 @@ To see `generateText` in action, check out [these examples](#examples).
         'Additional HTTP headers to be sent with the request. Only applicable for HTTP-based providers.',
     },
     {
-      name: 'maxToolRoundtrips',
+      name: 'maxSteps',
       type: 'number',
       isOptional: true,
       description:
-        'Maximum number of automatic roundtrips for tool calls. An automatic tool call roundtrip is another LLM call with the tool call results when all tool calls of the last assistant message have results. A maximum number is required to prevent infinite loops in the case of misconfigured tools. By default, it is set to 0, which will disable the feature.',
+        'Maximal number of sequential LLM calls (steps), e.g. when you use tool calls. A maximum number is required to prevent infinite loops in the case of misconfigured tools. By default, it is set to 1.',
```
Comment (nicoalbanese) [inline content/docs/07-reference/ai-sdk-core/01-generate-text.mdx:354]: should this be maximum?

File: content/docs/07-reference/ai-sdk-core/02-stream-text.mdx [hunk 2]
```diff
@@ -349,11 +349,11 @@ To see `streamText` in action, check out [these examples](#examples).
         'Additional HTTP headers to be sent with the request. Only applicable for HTTP-based providers.',
     },
     {
-      name: 'maxToolRoundtrips',
+      name: 'maxSteps',
       type: 'number',
       isOptional: true,
       description:
-        'Maximum number of automatic roundtrips for tool calls. An automatic tool call roundtrip is another LLM call with the tool call results when all tool calls of the last assistant message have results. A maximum number is required to prevent infinite loops in the case of misconfigured tools. By default, it is set to 0, which will disable the feature.',
+        'Maximal number of sequential LLM calls (steps), e.g. when you use tool calls. A maximum number is required to prevent infinite loops in the case of misconfigured tools. By default, it is set to 1.',
```
Comment (nicoalbanese) [inline content/docs/07-reference/ai-sdk-core/02-stream-text.mdx:356]: here again?

File: content/docs/07-reference/ai-sdk-ui/01-use-chat.mdx [hunk 3]
```diff
@@ -160,10 +160,10 @@ Allows you to easily create a conversational user interface for your chatbot app
         "An optional boolean that determines whether to send extra fields you've added to `messages`. Defaults to `false` and only the `content` and `role` fields will be sent to the API endpoint. If set to `true`, the `name`, `data`, and `annotations` fields will also be sent.",
     },
     {
-      name: 'maxToolRoundtrips',
+      name: 'maxSteps',
       type: 'number',
       description:
-        'React and SolidJS only. Maximal number of automatic roundtrips for tool calls. An automatic tool call roundtrip is a call to the server with the  tool call results when all tool calls in the last assistant  message have results. A maximum number is required to prevent infinite loops in the case of misconfigured tools. By default, it is set to 0, which will disable the feature.',
+        'React, Svelte and SolidJS only. Maximal number of backend calls to generate a response. A maximum number is required to prevent infinite loops in the case of misconfigured tools. By default, it is set to 1.',
```
Comment (nicoalbanese) [inline content/docs/07-reference/ai-sdk-ui/01-use-chat.mdx:166]: here too ("maximal")

File: content/examples/02-next-pages/10-tools/01-call-tool.mdx [hunk 4]
```diff
@@ -29,15 +29,15 @@ Some models allow developers to provide a list of tools that can be called at an
 
 Let's create a React component that imports the `useChat` hook from the `ai/react` module. The `useChat` hook will call the `/api/chat` endpoint when the user sends a message. The endpoint will generate the assistant's response based on the conversation history and stream it to the client. If the assistant responds with a tool call, the hook will automatically display them as well.
 
-We will use the `maxToolRoundtrips` to specify the maximum number of consecutive tool calls that can made before the model or the user responds with a text message. In this example, you will set it to `1` to allow for one round trip to happen.
+We will use the `maxSteps` to specify the maximum number of steps (i.e., LLM calls) that can be made to prevent infinite loops. In this example, you will set it to `2` to allow for one round trip to happen.
```
Comment (nicoalbanese) [inline content/examples/02-next-pages/10-tools/01-call-tool.mdx:32]: using "round trip" here. fine with me but just flagging

File: content/examples/03-node/03-tools/01-call-tool.mdx [hunk 5]
```diff
@@ -150,4 +150,8 @@ main().catch(console.error);
 
 ## Model Response
 
-When using tools, it's important to note that the model won't respond with the tool call results by default. This is because the model has technically already generated its response to the prompt: the tool call. Many use cases will require the model to summarise the results of the tool call within the context of the original prompt automatically. You can achieve this by [using `maxToolRoundtrips`](/examples/node/tools/call-tools-with-automatic-roundtrips) which will automatically send toolResults to the model to trigger another generation.
+When using tools, it's important to note that the model won't respond with the tool call results by default.
+This is because the model has technically already generated its response to the prompt: the tool call.
+Many use cases will require the model to summarise the results of the tool call within the context of the original prompt automatically.
+You can achieve this by [using `maxSteps`](/examples/node/tools/call-tools-with-automatic-roundtrips)
```
Comment (nicoalbanese) [inline content/examples/03-node/03-tools/01-call-tool.mdx:156]: does this link need to be replaced?

File: content/examples/03-node/03-tools/05-call-tools-with-automatic-roundtrips.mdx [hunk 6]
```diff
@@ -1,16 +1,16 @@
...
 
-# Call Tools with Roundtrips
+# Multi-step Tool Calls
 
 Models call tools to gather information or perform actions that are not directly available to the model.
 When tool results are available, the model can use them to generate another response.
 
-You can enable automatic roundtrips in `generateText` by setting the `maxToolRoundtrips` option to
+You can enable multi-step tool calls in `generateText` by setting the `maxSteps` option to
 a number greater than 0.
```
Comment (nicoalbanese) [inline content/examples/03-node/03-tools/05-call-tools-with-automatic-roundtrips.mdx:12]: this needs to be > 1, right?

File: content/examples/03-node/03-tools/index.mdx [hunk 7]
```diff
@@ -22,8 +22,8 @@ The following sections will guide you through tool use with Node.js and the Verc
       href: '/examples/node/tools/call-tools-in-parallel',
     },
     {
-      title: 'Call Tools with Automatic Roundtrips',
-      href: '/examples/node/tools/call-tools-with-automatic-roundtrips',
+      title: 'Multi-step Tool Calls',
+      href: '/examples/node/tools/tool-calls-with-automatic-roundtrips',
```
Comment (nicoalbanese) [inline content/examples/03-node/03-tools/index.mdx:26]: connected to previous point - check links are correct (doesn't seem in line with current name)

File: content/examples/03-node/03-tools/01-call-tool.mdx (hunk 5 above)
Comment (lgrammel) [reply content/examples/03-node/03-tools/01-call-tool.mdx:156]: wanted to keep it so we don't need to setup redirects

File: examples/sveltekit-openai/package.json [hunk 8]
```diff
@@ -3,7 +3,7 @@
   "version": "0.0.0",
   "private": true,
   "scripts": {
-    "dev": "vite dev",
+    "dev": "vite dev --port 3000",
```
Comment (nicoalbanese) [inline examples/sveltekit-openai/package.json:6]: did you mean to add this as part of this pr?

File: packages/ai/core/generate-text/generate-text-result.ts [hunk 9]
```diff
@@ -114,6 +116,53 @@ Additional response information.
     readonly response: LanguageModelResponseMetadataWithHeaders;
   }>;
 
+  /**
+Response information for every steps.
```
Comment (nicoalbanese) [inline packages/ai/core/generate-text/generate-text-result.ts:120]: for every "step"?

File: packages/ai/core/generate-text/generate-text-result.ts [hunk 10]
```diff
@@ -114,6 +116,53 @@ Additional response information.
...
+    readonly usage: LanguageModelUsage;
+
+    /**
+Warnings from the model provider (e.g. unsupported settings)
+ */
+    readonly warnings: CallWarning[] | undefined;
+
+    /**
+Logprobs for the completion.
+`undefined` if the mode does not support logprobs or if was not enabled.
```
Comment (nicoalbanese) [inline packages/ai/core/generate-text/generate-text-result.ts:156]: "or if **it** was not enabled"

File: packages/ai/core/generate-text/generate-text.ts [hunk 11]
```diff
@@ -69,7 +70,7 @@ If set and supported by the model, calls will generate deterministic results.
 @param abortSignal - An optional abort signal that can be used to cancel the call.
 @param headers - Additional HTTP headers to be sent with the request. Only applicable for HTTP-based providers.
 
-@param maxToolRoundtrips - Maximal number of automatic roundtrips for tool calls.
+@param maxSteps - Maximal number of sequential LLM calls (steps), e.g. when you use tool calls.
```
Comment (nicoalbanese) [inline packages/ai/core/generate-text/generate-text.ts:73]: Maximal -> maximum

File: examples/sveltekit-openai/package.json (hunk 8 above)
Comment (lgrammel) [reply examples/sveltekit-openai/package.json:6]: i felt it was not worth separating out

File: packages/ai/core/generate-text/generate-text.ts [hunk 12]
```diff
@@ -126,9 +128,20 @@ A maximum number is required to prevent infinite loops in the
 case of misconfigured tools.
 
 By default, it's set to 0, which will disable the feature.
+
+@deprecated Use `maxSteps` instead (which is `maxToolRoundtrips` + 1).
      */
     maxToolRoundtrips?: number;
 
+    /**
+Maximal number of sequential LLM calls (steps), e.g. when you use tool calls. Must be at least 1.
```
Comment (nicoalbanese) [inline packages/ai/core/generate-text/generate-text.ts:137]: Maximal -> Maximum

**Review by nicoalbanese — APPROVED** [review]

**Pull request merged**